python/src/aiconfig/eval/lib.py

```diff
@@ -223,23 +264,23 @@ def _zip_inputs_outputs(outputs: list[TextOutput]):
 
 def user_test_suite_outputs_only_to_eval_params_list(
     test_suite: UserTestSuiteOutputsOnly,
-) -> DatasetEvaluationParams[TextInput, TextOutput]:
+) -> DatasetEvaluationParams[TextBasedInputDatum, TextOutput]:
     """
     Example: [("the_output_is_world", brevity)] -> [SampleEvaluationParams(None, "the_output_is_world", brevity)
     """
     return [SampleEvaluationParams(input_sample=None, output_sample=TextOutput(output_datum), metric=metric) for output_datum, metric in test_suite]
 
 
-async def run_aiconfig_helper(runtime: AIConfigRuntime, prompt_name: str, question: str) -> Result[str, str]:
-    params = {
-        "the_query": question,
-    }
+async def run_aiconfig_on_text_based_input(runtime: AIConfigRuntime, prompt_name: str, params: TextBasedInputDatum) -> Result[str, str]:
+    def _get_params_for_aiconfig(params: TextBasedInputDatum) -> dict[str, str]:
+        match params.value:
+            case str(input_text):
+                return {"the_query": input_text}
+            case frozendict():
```

Comment:
Is this checking if it's an empty frozendict right now (i.e. should this be instanceof)?

Reply:
Very good question - I struggled with some of the subtleties of py3 pattern matching. One thing I learned is it doesn't really work with NewType, which doesn't, um, actually really make a new type :D.


To answer your question, this matches all frozendict instances. I don't fully understand how this works for primitives and non-record container types like this, but I have confirmed this works. 

The main thing about match...case is that pyright can check for exhaustive matching. So if this were only matching empty frozendicts, pyright would tell me I havent matched the entire input type.